processor/filestorage/s3.go

```diff
@@ -0,0 +1,61 @@
+package filestorage
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	derrors "github.com/skroutz/downloader/processor/errors"
+	"os"
+)
+
+type AWSS3 struct {
+	bucket   string
+	uploader *s3manager.Uploader
+	S3Client *s3.S3
+}
+
+func NewAWSS3(bucket string) *AWSS3 {
+	s3Session, err := session.NewSession()
+	if err != nil {
+		// FIXME
+		return &AWSS3{"", nil, nil} //derrors.E("creating AWS S3 session", err)
```

Comment:
I guess this should return `nil` if we cant create a session and the caller should handle the error itself.